core/trino-main/src/main/java/io/trino/sql/analyzer/FeaturesConfig.java

```diff
@@ -135,6 +135,15 @@
     private DataSize filterAndProjectMinOutputPageSize = DataSize.of(500, KILOBYTE);
     private int filterAndProjectMinOutputPageRowCount = 256;
     private int maxGroupingSets = 2048;
+    /*
+     * We leave join pushdown disabled by default as this is safer option.
+     * Pushing down a join which substantially increases the row count vs
+     * sizes of left anf right table separately, may incur huge cost both
+     * in terms of performance and money due to an increased network traffic.
+     *
+     * As soon as we have cost based logic to determine join pushdown feasibility we
+     * will change the default to automatic.
+     */
```

Comment:
It's OK for now, but we will actually want to enable it by default on the engine side, and then disable by default on the connector side.

Thus, the engine side should be simplified to be true/false toggle, and _true_ by default.

Reply:
i see it's being removed in next commit. Thanks for having this in.

Reply:
With new approach we should probably drop `JoinPushdownMode` altogether and just have `enable-join-pushdown` config property (and matching session property).
WDYT @findepi

Reply:
Yes, and BTW this is what i was trying to express. You just did it way better

Reply:
You are very kind :)